### PR TITLE
ci: add gunicorn framework tests

### PR DESCRIPTION
**Pull request by P403n1x87**
## Description

Adding the gunicorn test suite to the framework tests to better test support for gevent.

## Checklist
- [ ] Followed the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) when writing a release note.
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

File: .github/workflows/test_frameworks.yml [hunk 1]
```diff
@@ -416,3 +416,31 @@ jobs:
           sed -i'' "s/test_langs/langs/g" tests/test_units/test_basic_app.py
       - name: Run tests
         run: nosetests
+
+  gunicorn-testsuite-20_1_0:
+    name: gunicon 20.1.0
```
Comment (Kyle-Verhoog) [inline .github/workflows/test_frameworks.yml:421]:
```suggestion
    name: gunicorn 20.1.0
```

File: .github/workflows/test_frameworks.yml [hunk 2]
```diff
@@ -416,3 +416,31 @@ jobs:
...
+    env:
+      DD_TESTING_RAISE: true
+      # PYTHONPATH: ../ddtrace/tests/debugging/exploration/
+    defaults:
+      run:
+        working-directory: gunicorn
+    steps:
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
```
Comment (Kyle-Verhoog) [inline .github/workflows/test_frameworks.yml:432]:
```suggestion
          python-version: '3.10'
```

if we can

File: .github/workflows/test_frameworks.yml [hunk 3]
```diff
@@ -416,3 +416,31 @@ jobs:
           sed -i'' "s/test_langs/langs/g" tests/test_units/test_basic_app.py
       - name: Run tests
         run: nosetests
+
+  gunicorn-testsuite-20_1_0:
+    name: gunicon 20.1.0
+    runs-on: "ubuntu-latest"
+    env:
+      DD_TESTING_RAISE: true
+      # PYTHONPATH: ../ddtrace/tests/debugging/exploration/
```
Comment (Kyle-Verhoog) [inline .github/workflows/test_frameworks.yml:425]: should we uncomment this line?

Comment (P403n1x87) [reply .github/workflows/test_frameworks.yml:425]: Not at this time unfortunately 🙁 . Exploration tests cause the test suite run to freeze, which I think is what we're trying to fix in the long-term.

File: .github/workflows/test_frameworks.yml (hunk 2 above)
Comment (P403n1x87) [reply .github/workflows/test_frameworks.yml:432]: gunicorn 20.1.0 doesn't seem to support 3.10 🙁

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by Yun-Kim — APPROVED** [review]

**Pull request merged**